### PR TITLE
Enable "Shaded Wireframe" scene view mode in HDRP & URP [Need C++ PR not backport on 2020.1

### PR DESCRIPTION
**Pull request by aras-p**
### Purpose of this PR
Why is this PR needed, what hard problem is it solving/fixing?

Depends on C++ PR here: https://ono.unity3d.com/unity/unity/pull-request/105681/_/graphics/srp/textured-wire

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by phi-lira** [comment]
We should we bump the `unityRelease` in package.json to match the version of Unity that contains this new API.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 1]
```diff
@@ -2595,6 +2595,9 @@ void Callback(CommandBuffer c, HDCamera cam)
             // We need to make sure the viewport is correctly set for the editor rendering. It might have been changed by debug overlay rendering just before.
             cmd.SetViewport(hdCamera.finalViewport);
 
+            if (camera.cameraType == CameraType.SceneView)
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:2598]: The test should check if we are the current texturewire mode no ?

var sv = UnityEditor.SceneView.sceneViews[i] as UnityEditor.SceneView;
sv.cameraMode.drawMode != UnityEditor.DrawCameraMode.TexturedWire 

(but not sure how to get the current scene view as there is multiple one).

Comment (aras-p) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:2598]: Yeah looks like scene view render mode of camera is not exposed to scripts, so checks could only be done in a very roundabout way. `RenderWireOverlay` internally does nothing if the mode is not TexturedWire.

Comment (sebastienlagarde) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:2598]: ok, good to know

**Comment by sebastienlagarde** [comment]
C++ PR have landed now, let's merge it (waiting for Yamato :) )

**Comment by sebastienlagarde** [comment]
Yamato is green and the PR have landed. Merging

**Comment by aras-p** [comment]
@sebastienlagarde @phi-lira: since this now requires a new API that is only since 2020.2, does that mean the usages should get wrapped into some sort of `#if` check? Or are master branches of HDRP & URP only compatible with 2020.2 anyway at this point?

**Comment by sebastienlagarde** [comment]
Do we plan a backport of it?  (if possible would be nice)
If no backport is plan, then we don't care, the code will only be in 10.x package  and only for 2020.2.

Otherwise, we can backport this PR in 9.x and 8.x when the backport land.

**Comment by phi-lira** [comment]
@aras-p It would also be nice to have a wrap with unity version macro. 

"If no backport is plan, then we don't care, the code will only be in 10.x package and only for 2020.2." a good point of wrapping in macros for 2020.2 f.ex, is that when bisecting an issue we can run older package with recent version of Unity. This helps bisecting a lot when we suspect the regression is in Unity side.

**Comment by sebastienlagarde** [comment]
>a good point of wrapping in macros for 2020.2 f.ex, is that when bisecting an issue we can run older
This doesn't help you here

if you have #ifdef UNITY_VERSION_2020_2_OR_NEWER, it is define since we have branch off 2020.2.
So the guard is useless for a bisecting.

I prefer to avoid to pollute code with unecesary #ifdef

**Pull request merged**